akka-projection-kafka/src/main/scala/akka/projection/kafka/javadsl/KafkaSourceProvider.scala

```diff
@@ -9,24 +9,34 @@ import akka.kafka.ConsumerSettings
 import akka.projection.kafka.GroupOffsets
 import akka.projection.kafka.internal.KafkaSourceProviderImpl
 import akka.projection.kafka.internal.MetadataClientAdapterImpl
-import akka.projection.javadsl.SourceProvider
+import akka.projection.javadsl
+import akka.projection.kafka.internal.KafkaSourceProviderSettings
 import org.apache.kafka.clients.consumer.ConsumerRecord
 
 object KafkaSourceProvider {
 
   /**
-   * Create a [[SourceProvider]] that resumes from externally managed offsets
+   * Create a [[KafkaSourceProvider]] that resumes from externally managed offsets
    */
   def create[K, V](
       system: ActorSystem[_],
       settings: ConsumerSettings[K, V],
-      topics: java.util.Set[String]): SourceProvider[GroupOffsets, ConsumerRecord[K, V]] = {
+      topics: java.util.Set[String]): KafkaSourceProvider[K, V] = {
     import akka.util.ccompat.JavaConverters._
     new KafkaSourceProviderImpl[K, V](
       system,
       settings,
       topics.asScala.toSet,
-      new MetadataClientAdapterImpl(system, settings))
+      new MetadataClientAdapterImpl(system, settings),
+      KafkaSourceProviderSettings(system),
+      readOffsetDelay = None)
   }
 
 }
+
+abstract class KafkaSourceProvider[K, V]
+    extends javadsl.SourceProvider[GroupOffsets, ConsumerRecord[K, V]]
+    with javadsl.VerifiableSourceProvider[GroupOffsets, ConsumerRecord[K, V]]
+    with javadsl.MergeableOffsetSourceProvider[GroupOffsets.TopicPartitionKey, GroupOffsets, ConsumerRecord[K, V]] {
+  def withReadOffsetDelay(delay: java.time.Duration): KafkaSourceProvider[K, V]
+}
```

Comment:
I don't think we should have this at this level. Enough with config property to rule all KafkaSourceProvider instances.

Reply:
Ok. I was trying to be consistent with the projection settings, but you're probably right.